esmvaltool/utils/cmorizers/obs/utilities.py

```diff
@@ -126,26 +126,30 @@ def read_cmor_config(cmor_config):
 def save_variable(cube, var, outdir, attrs, **kwargs):
     """Saver function."""
     # CMOR standard
-    cube_time = cube.coord('time')
-    reftime = Unit(cube_time.units.origin, cube_time.units.calendar)
-    dates = reftime.num2date(cube_time.points[[0, -1]])
-    if len(cube_time.points) == 1:
-        year = str(dates[0].year)
-        time_suffix = '-'.join([year + '01', year + '12'])
-    else:
-        date1 = str(dates[0].year) + '%02d' % dates[0].month
-        date2 = str(dates[1].year) + '%02d' % dates[1].month
-        time_suffix = '-'.join([date1, date2])
-
-    file_name = '_'.join([
+    try:
+        cube_time = cube.coord('time')
+        reftime = Unit(cube_time.units.origin, cube_time.units.calendar)
```

Comment:
Use an `else` clause for the code that does not need to be in the try block, i.e. everthing after the first line. This is recommended by [PEP8](https://www.python.org/dev/peps/pep-0008/#programming-recommendations)

> Additionally, for all try/except clauses, limit the try clause to the absolute minimum amount of code necessary. Again, this avoids masking bugs.